concent_api/middleman/middleman_server.py

```diff
@@ -108,26 +115,43 @@ def _close_middleman(self) -> None:
 
     async def _handle_concent_connection(self, reader, writer) -> None:
         try:
-            data = await reader.readuntil()
+            data = await handle_frame_receive_async(reader)
             remote_address = writer.get_extra_info('peername')
             print("Received %r from %r" % (data, remote_address))
-            await self._respond_to_user(data, writer)
+            serialized_message = self._prepare_response(data)
+            await self._respond_to_user(serialized_message, writer)
         except Exception as exception:  # pylint: disable=broad-except
             crash_logger.error(
                 f"Exception occurred: {exception}, Traceback: {traceback.format_exc()}"
             )
             raise
 
+    @staticmethod
+    def _prepare_response(data):
```

Comment:
You can name parameter `serialized_message`, then you can remove line `serialized_message = data`.

Reply:
Hmm it's not necessarily `serialized`. Maybe just remove line `serialized_message = data`, and use/return `data`?

Reply:
Would be ok too.